rewrite accessibility section on invokers (#1078)
openui/open-ui: site/src/pages/components/invokers.explainer.mdx

that, when connected to those elements via `commandfor`, buttons will
implicitly recieve `aria-haspopup` of the matching type.

</details>

<br/><br/>

If `commandfor=` does not point to an element connected to the DOM then this
attribute must not be present.

Buttons with these popover commands will also have focus restored to the
button when the popover closes. This will require the popover keeping a
reference to the opening button so that it can return the focus to that button.

<details>
<summary>Why?</summary>

It is important for keyboard users (or other users who rely on focus) to not
have their focus state lost when performing actions on the page. If the page
shows a new piece of navigable UI, and shifts the user's focus to that UI,
then it is important to restore focus when that UI closes, so that the user's
journey is maintained.

</details>

<br/><br/>

<details>
<summary>Other considerations not explicitly proposed.</summary>

##### aria-pressed

Given elements will have `aria-expanded`, adding `aria-pressed` would be
confusing or redundant, and as such won't be proposed for these buttons.

##### aria-controls

> While `aria-controls` attempts to establish a similar style of relationship to

Reviewer: this isn't really accurate - re: parent/child relationship.  i'm not sure why this is mentioned at all if it's not being used / was never implemented for popovers?

Author: I tried to add these extra bits as somewhat of an FAQ to field questions of "why hasn't this particular property been implemented", to have some written down rationale that people can refer to when those questions arise.

Reviewer: ok, that makes sense.  

so then it's more of a way this is described then - since it's not commonly a parent/child relationship - but more of an association between related elements

Author: Would it be better to point out instead that `aria-details` has better support, and applying both is redundant?

Reviewer: yah that could work.  so something like

aria-controls is not used due to lack of support among screen readers, and even if it was supported it would be redundant to the association already created due to the implicit aria-details relationship already created.

`aria-details`, it is largely intented for parent-child relationships. Buttons
that open popovers do not necessarily represent a parent-child relationship.

</details>

<br/><br/>

#### Buttons with `command=show-modal`

> Buttons with this dialog commands will implicitly receive `aria-haspopup=dialog`.

Reviewer: same comments as before.  the intent for the haspopup was more to help users know what types of popups were invoked from a combobox.  i really am trying to get ARIA back on track for that being the primary purpose of this attribute, i really dont want to see this get used for buttons and further cement a practice this wasn't originally intended for.

implicitly recieve `aria-haspopup` of the matching type.

</details>

<br/><br/>

Buttons with this dialog command will also have focus restored to the button
when the dialog closes. This will require the dialog keeping a reference to the
opening button so that it can return the focus to that button.

<details>
<summary>Why?</summary>

It is important for keyboard users (or other users who rely on focus) to not
have their focus state lost when performing actions on the page. If the page
shows a new piece of navigable UI, and shifts the user's focus to that UI,
then it is important to restore focus when that UI closes, so that the user's
journey is maintained.

</details>

<br/><br/>

<details>
<summary>Other considerations not explicitly proposed.</summary>

##### aria-details

Exposing an `aria-details` relationship between the button and the dialog is
redundant. These relationships are useful only when the related element is
open and present on the page and the invoking button is still navigable. A
modal dialog renders the rest of the page inert meaning that it won't be
possible to navigate to the invoking button.

##### aria-pressed

> The `aria-pressed` state is the incorrect type of association for elements

Reviewer: it's weird this says aria-expanded would be the correct state to expose, but in the next section also talks about how the state is redundant.

which conditionally appear on the page. The correct type of association would
be `aria-expanded`...

##### aria-expanded

Similar to `aria-details`, the `aria-expanded` state redundant, given the
dialog is opened as modal. For the user to have navigated to a button that
opens a modal dialog, the dialog must not be open. If the dialog is open then
the button will be inert, therefore non-navigable, therefore it is redundant
to provide it an implicit `aria-expanded=true` state also.

##### aria-controls

> While `aria-controls` attempts to establish a similar style of relationship to

Reviewer: similar as before, the description here is not accurate, but also just doesn't seem worth mentioning if the only reason to do so is to say "it's not used"

modal dialog renders the rest of the page inert meaning that it won't be
possible to navigate to the invoking button.

##### aria-controls

Similar to `aria-details`, `aria-controls` is not useful for a dialog that

> is closed, and non-navigable for a dialog that is open as modal, so it would be
> useless to add.

Reviewer: wording nit to not be redundant in saying it 'is not useful' and then 'it would be useless...."

```suggestion
is closed, and non-navigable for a dialog that is open as modal, so it would 
provide no benefit to add.
```

the button will be inert, therefore non-navigable, therefore it is redundant
to provide it an implicit `aria-expanded=true` state also.

##### aria-haspopup=dialog

> When the `commandfor=` element is a `<dialog>`, adding aria-haspopup=dialog is
> a potentially viable attribute to add. Some Assistive Technologies will announce

Reviewer: ```suggestion
When the `commandfor=` element is a `<dialog>`, `aria-haspopup=dialog` might seem
a potentially viable attribute to add. Some Assistive Technologies will announce
```

that the button will open a dialog, for example
`<button aria-haspopup=dialog>Delete</button>` may be announced by a screen
reader as "Delete, button, opens dialog". However, buttons that open dialogs

> rarely come with additional visual treatment, and so the lack of a visual
> analog means that using this attribute likely provides surplus information

Reviewer: ```suggestion
rarely come with additional visual treatment, and so the [lack of a visual
analog means that using this attribute likely provides surplus information](https://w3c.github.io/aria/#h-note-50)
```
